perf/reader.go

```diff
@@ -321,19 +335,33 @@ func (pr *Reader) ReadInto(rec *Record) error {
 
 	for {
 		if len(pr.epollRings) == 0 {
-			nEvents, err := pr.poller.Wait(pr.epollEvents, pr.deadline)
-			if err != nil {
-				return err
-			}
+			if pr.overWritable {
```

Comment:
In the current implementation, `SetDeadline` has no effect on an overWritable reader.

```go
r.SetDeadline(time.Now().Add(time.Second))
r.Read() // returns io.EOF without blocking
```

I think `SetDeadline` should always work consistently. You would use the reader like so:

```go
// this can happen once after creation of r, it just has to
// be Now() or earlier. See Poller.Wait.
r.SetDeadline(time.Now())
// call repeatedly
r.Read() // returns ErrDeadlineExceeded now
```

This means that we don't need this change and the one in the `err == errEOR` branch.

P.S.: I realize now that the kernel might do funky things wrt epoll and backwards written rings. Still worth a try.

Reply:
The poller will wait forever only if there is *no deadline set*. If you set a deadline and there is nothing to read, you will get ErrDeadlineExceeded.

Reply:
Maybe this is helpful: https://man7.org/linux/man-pages/man2/epoll_wait.2.html

> Specifying a
       timeout of -1 causes epoll_wait() to block indefinitely, while
       specifying a timeout equal to zero cause epoll_wait() to return
       immediately, even if no events are available.